src/pgen/cluster/agn_triggering.cpp

```diff
@@ -0,0 +1,626 @@
+        },
+        "agn_accretion_rate"));
+  }
+  hydro_pkg->UpdateParam(parthenon::hist_param_key, hst_vars);
+
+  hydro_pkg->AddParam<AGNTriggering>("agn_triggering", *this);
+}
+
+// Compute Cold gas accretion rate within the accretion radius for cold gas triggering
+// and simultaneously remove cold gas (updating conserveds and primitives)
+template <typename EOS>
+void AGNTriggering::ReduceColdMass(parthenon::Real &cold_mass,
+                                   parthenon::MeshData<parthenon::Real> *md,
+                                   const parthenon::Real dt, const EOS eos) const {
+  using parthenon::IndexDomain;
+  using parthenon::IndexRange;
+  using parthenon::Real;
+
+  auto hydro_pkg = md->GetBlockData(0)->GetBlockPointer()->packages.Get("Hydro");
+
+  // Grab some necessary variables
+  const auto &prim_pack = md->PackVariables(std::vector<std::string>{"prim"});
+  const auto &cons_pack = md->PackVariables(std::vector<std::string>{"cons"});
+  IndexRange ib = md->GetBlockData(0)->GetBoundsI(IndexDomain::interior);
+  IndexRange jb = md->GetBlockData(0)->GetBoundsJ(IndexDomain::interior);
+  IndexRange kb = md->GetBlockData(0)->GetBoundsK(IndexDomain::interior);
+  const auto nhydro = hydro_pkg->Param<int>("nhydro");
+  const auto nscalars = hydro_pkg->Param<int>("nscalars");
+
+  const Real accretion_radius2 = pow(accretion_radius_, 2);
+
+  // Reduce just the cold gas
+  const auto units = hydro_pkg->Param<Units>("units");
+  const Real mean_molecular_mass_by_kb = mean_molecular_mass_ / units.k_boltzmann();
+
+  const Real cold_temp_thresh = cold_temp_thresh_;
+  const Real cold_t_acc = cold_t_acc_;
+
+  const bool remove_accreted_mass = remove_accreted_mass_;
+
+  Real md_cold_mass = 0;
+
+  parthenon::par_reduce(
+      parthenon::loop_pattern_mdrange_tag, "AGNTriggering::ReduceColdGas",
+      parthenon::DevExecSpace(), 0, cons_pack.GetDim(5) - 1, kb.s, kb.e, jb.s, jb.e, ib.s,
+      ib.e,
+      KOKKOS_LAMBDA(const int &b, const int &k, const int &j, const int &i,
+                    Real &team_cold_mass) {
+        auto &cons = cons_pack(b);
+        auto &prim = prim_pack(b);
+        const auto &coords = cons_pack.GetCoords(b);
+
+        const parthenon::Real r2 =
+            pow(coords.Xc<1>(i), 2) + pow(coords.Xc<2>(j), 2) + pow(coords.Xc<3>(k), 2);
+        if (r2 < accretion_radius2) {
+
+          const Real temp =
+              mean_molecular_mass_by_kb * prim(IPR, k, j, i) / prim(IDN, k, j, i);
+
+          if (temp <= cold_temp_thresh) {
+
+            const Real cell_cold_mass = prim(IDN, k, j, i) * coords.CellVolume(k, j, i);
+            team_cold_mass += cell_cold_mass;
+
+            const Real cell_delta_rho = -prim(IDN, k, j, i) / cold_t_acc * dt;
+
+            if (remove_accreted_mass) {
+              AddDensityToConsAtFixedVelTemp(cell_delta_rho, cons, prim, eos, k, j, i);
+              // Update the Primitives
+              eos.ConsToPrim(cons, prim, nhydro, nscalars, k, j, i);
```

Comment:
Reminder to myself: double check this is the intended behavior (in terms of where this function is called in the task list)

Reply:
POTENTIAL BUG (should discuss today)
If we update the variables here, we should also do it in the ghost zones as the `interior` info is currently not communicated any more.

Reply:
I've updated both the cold gas triggering and Bondi accretion to remove mass from the entire grid, not just interior grid. The reductions are still done over the interior grid so as to not double count gas.